blsSignatures/blsSignatures.go

```diff
@@ -64,15 +68,15 @@ func PublicKeyFromPrivateKey(privateKey PrivateKey) (PublicKey, error) {
 //
 // For a proof that this is sufficient, see Theorem 1 in
 // Ristenpart & Yilek, "The Power of Proofs-of-Possession: ..." from EUROCRYPT 2007.
-func KeyValidityProof(pubKey *bls12381.PointG2, privateKey PrivateKey) (Signature, error) {
-	g2 := bls12381.NewG2()
-	return signMessage2(privateKey, g2.ToBytes(pubKey), true)
+func KeyValidityProof(pubKey *bls12381.G2Affine, privateKey PrivateKey) (Signature, error) {
+	message := pubKey.Bytes()
+	return signMessage2(privateKey, message[:], true)
```

Comment:
We should add a test with a hardcoded private key and public key (including a validity proof) generated from the previous version's code, and make sure the private key corresponds to the expected public key, and also ensure the validity proof passes.

Reply:
Done.